scipy/stats/tests/test_sensitivity_analysis.py

```diff
@@ -0,0 +1,288 @@
+            stats.uniform(loc=-np.pi, scale=2*np.pi)
+        ]
+
+        A, B = sample_A_B(n=n, dists=dists, random_state=rng)
+        AB = sample_AB(A=A, B=B)
+
+        func = {
+            'f_A': f_ishigami(A).reshape(1, -1),
+            'f_B': f_ishigami(B).reshape(1, -1),
+            'f_AB': f_ishigami(AB).reshape((3, 1, -1))
+        }
+
+        res = sobol_indices(
+            func=func, n=n,
+            dists=dists,
+            random_state=rng
+        )
+        assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
+
+        res = sobol_indices(
+            func=func, n=n,
+            random_state=rng
+        )
+        assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
+
+    def test_method(self, ishigami_ref_indices):
+        def jansen_sobol(f_A, f_B, f_AB):
+            """Jansen for S and Sobol' for St.
+
+            From Saltelli2010, table 2 formulations (c) and (e)."""
+            var = np.var([f_A, f_B], axis=(0, -1))
+
+            s = (var - 0.5*np.mean((f_B - f_AB)**2, axis=-1)) / var
+            st = np.mean(f_A*(f_A - f_AB), axis=-1) / var
+
+            return s.T, st.T
+
+        rng = np.random.default_rng(28631265345463262246170309650372465332)
+        res = sobol_indices(
+            func=f_ishigami, n=4096,
+            dists=self.dists,
+            method=jansen_sobol,
+            random_state=rng
+        )
+
+        assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
+        assert_allclose(res.total_order, ishigami_ref_indices[1], atol=1e-2)
+
+    def test_normalization(self, ishigami_ref_indices):
+        rng = np.random.default_rng(28631265345463262246170309650372465332)
+        res = sobol_indices(
+            func=lambda x: f_ishigami(x) + 1000, n=4096,
+            dists=self.dists,
+            random_state=rng
+        )
+
+        assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
+        assert_allclose(res.total_order, ishigami_ref_indices[1], atol=1e-2)
+
+    def test_constant_function(self, ishigami_ref_indices):
+
+        def f_ishigami_vec_const(x):
+            """Output of shape (3, n)."""
+            res = f_ishigami(x)
+            return res, res * 0 + 10, res
+
+        rng = np.random.default_rng(28631265345463262246170309650372465332)
+        res = sobol_indices(
+            func=f_ishigami_vec_const, n=4096,
+            dists=self.dists,
+            random_state=rng
+        )
+
+        ishigami_vec_indices = [
+                [ishigami_ref_indices[0], [0, 0, 0], ishigami_ref_indices[0]],
+                [ishigami_ref_indices[1], [0, 0, 0], ishigami_ref_indices[1]]
+        ]
+
+        assert_allclose(res.first_order, ishigami_vec_indices[0], atol=1e-2)
+        assert_allclose(res.total_order, ishigami_vec_indices[1], atol=1e-2)
+
+    @pytest.mark.xfail_on_32bit("Can't create large array for test")
+    def test_more_converged(self, ishigami_ref_indices):
```

Comment:
This suggests the need for a `batch` parameter as an enhancement. Instead of evaluating the function at all `n` points at a time, evaluate the function at `batch` points at a time. (Yes, loop and concatenate. If `batch` is ~100 or so, you get most of the speed of vectorizing the computation but far less memory usage.)